### DISCUSSION TITLE
eMMC regression in Linux master and stable breaks boot-emmc updates

### DISCUSSION THREAD

**Discussion by jluebbe** [rauc/rauc#1363] [Announcements]
A change in the MMC stack in the Linux kernel introduced a regression when changing the active eMMC boot partition after an update. In RAUC, this affects bootloader updates using the `boot-emmc` slot type. Sometimes a check in RAUC detects the problem and aborts the installation with a message similar to `Installation error: Failed updating slot bootloader.0: Toggling the boot partition failed!`. If RAUC doesn't detect the issue, it can happen that the old partition is still active after the update completes.

The problematic change was identified [here](https://lore.kernel.org/all/Ze8OXcCA_BCN2MVE@nuoska/T/#m21567eb27e088076299e832c3aa75349b958ff5c). A fix is available [here](https://lore.kernel.org/all/20240313133744.2405325-1-mikko.rapeli@linaro.org/) and has been back-ported to the affected stable branches.

The following versions are affected: v6.8…v6.8.2, v6.7.3…v6.7.11, v6.6.15…v6.6.23, v6.1.76…v6.1.83, v5.15.149…v5.15.153, v5.10.210…v5.10.214, v5.4.269…v5.4.273

The relevant commits are...
- master: introduced with https://github.com/torvalds/linux/commit/4d0c8d0aef6355660b6775d57ccd5d4ea2e15802 in v6.8-rc
  - fixed with https://github.com/torvalds/linux/commit/0cdfe5b0bf295c0dee97436a8ed13336933a0211 in v6.9-rc2
- stable-6.8: introduced during development phase
  - fixed with https://github.com/gregkh/linux/commit/d7b24b0d80d4799c4ed2906e99de41a0453aa659 in v6.8.3
- stable-6.7: backported as https://github.com/gregkh/linux/commit/eed9119f8f8e8fbf225c08abdbb58597fba807e0 in v6.7.3
  - fixed with https://github.com/gregkh/linux/commit/59169ee38d7b8f6358a869b8f0bb86b1ec5af5c7 in v6.7.12
- stable-6.6: backported as https://github.com/gregkh/linux/commit/1653a8102868264f3488c298a9f20af2add9a288 in v6.6.15
  - fixed with https://github.com/gregkh/linux/commit/35ee8529eee76bd858e279e7f751734b0d5bbc4b in v6.6.24
- stable-6.1: backported as https://github.com/gregkh/linux/commit/c4edcd134bb72b3b0acc884612d624e48c9d057f in v6.1.76
  - fixed with https://github.com/gregkh/linux/commit/547f4afaced0901d1d4c4bd64415dc1223f5fff8 in v6.1.84
- stable-5.15: backported as https://github.com/gregkh/linux/commit/50b8b7a22e90bab9f1949b64a88ff17ab10913ec in v5.15.149
  - fixed with https://github.com/gregkh/linux/commit/0ecd5689767b9bc85f1423b62481ba8c4a3f440d in v5.15.154
- stable-5.10: backported as https://github.com/gregkh/linux/commit/59020bf0999ff7da8aedcd00ef8f0d75d93b6d20 in v5.10.210
  - fixed with https://github.com/gregkh/linux/commit/bce3a98352eabb3341f2cc74c05125f85bc97158 in v5.10.215
- stable-5.4: backported as https://github.com/gregkh/linux/commit/f49f9e802785291149bdc9c824414de4604226b4 in v5.4.269
  - fixed with https://github.com/gregkh/linux/commit/fd20d84bae882291425254e87ad9698f5b62a000 in v5.4.274

**Comment by a3f** [comment]
The fix is now mainline:
* master: Fixed with https://github.com/torvalds/linux/commit/0cdfe5b0bf295c0dee97436a8ed13336933a0211 in v6.9-rc2

and should soon make it into stable releases. I'll update this comment when that happens.

**Reply by jluebbe** [reply]
Fixes have now landed in some stable releases: v6.8.3, v6.7.12, v6.6.24 and v6.1.84. See the updated post above for the exact commits.

**Reply by a3f** [reply]
stable-5.15: fixed with https://github.com/gregkh/linux/commit/0ecd5689767b9bc85f1423b62481ba8c4a3f440d in v5.15.154

**Reply by a3f** [reply]
* stable-5.10: fixed with https://github.com/gregkh/linux/commit/bce3a98352eabb3341f2cc74c05125f85bc97158 in v5.10.215
* stable-5.4: fixed with https://github.com/gregkh/linux/commit/fd20d84bae882291425254e87ad9698f5b62a000 in v5.4.274
